dbt/adapters/athena/python_submissions.py

```diff
@@ -0,0 +1,123 @@
+import time
+from functools import cached_property
```

Comment:
this doesn't work for 3.7. you can consider to use `functools.lru_cache` that could work too in a similar way and should be available in 3.7.

Reply:
Updated to use `lru_cache` as mentioned